mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -982,45 +1275,13 @@ private class LogisticAggregator(
             "coefficients only supports dense vector" +
               s"but got type ${bcCoefficients.value.getClass}.")
       }
-      val localGradientSumArray = gradientSumArray
-
-      val featuresStd = bcFeaturesStd.value
-      numClasses match {
-        case 2 =>
-          // For Binary Logistic Regression.
-          val margin = - {
-            var sum = 0.0
-            features.foreachActive { (index, value) =>
-              if (featuresStd(index) != 0.0 && value != 0.0) {
-                sum += coefficientsArray(index) * (value / featuresStd(index))
-              }
-            }
-            sum + {
-              if (fitIntercept) coefficientsArray(numFeatures) else 0.0
-            }
-          }
-
-          val multiplier = weight * (1.0 / (1.0 + math.exp(margin)) - label)
-
-          features.foreachActive { (index, value) =>
-            if (featuresStd(index) != 0.0 && value != 0.0) {
-              localGradientSumArray(index) += multiplier * (value / featuresStd(index))
-            }
-          }
-
-          if (fitIntercept) {
-            localGradientSumArray(numFeatures) += multiplier
-          }
 
-          if (label > 0) {
-            // The following is equivalent to log(1 + exp(margin)) but more numerically stable.
-            lossSum += weight * MLUtils.log1pExp(margin)
-          } else {
-            lossSum += weight * (MLUtils.log1pExp(margin) - margin)
-          }
-        case _ =>
-          new NotImplementedError("LogisticRegression with ElasticNet in ML package " +
-            "only supports binary classification for now.")
+      if (multinomial) {
+        multinomialUpdateInPlace(features, weight, label, coefficientsArray, gradientSumArray,
+          bcFeaturesStd.value, numFeaturesPlusIntercept)
```

Comment:
For consistency, I'll suggest that you either return `loss` explicitly from `multinomialUpdateInPlace` and `binaryUpdateInPlace` functions and sum them into `lossSum` to those two private functions don't use any variable in the class or you remove `coefficientsArray`, `gradientSumArray`, `bcFeaturesStd.value`, and `numFeaturesPlusIntercept` from the functions.